src/io.d

```diff
@@ -51,6 +53,7 @@ struct Input
             col = 0;
         }
         index_ += text[index_ .. $].stride;
+        ++offset;
```

Comment:
Should work because Xtext is already implemented multi line ending aware.
They only take NL into consideration as line ending, but every character counts into offset. 
See https://github.com/eclipse/xtext-core/blob/0fa87d4ba010cd11e7e90c587208e19cc30d808d/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/Document.java#L42